server/api/users/users.services.js

```diff
@@ -1,12 +1,31 @@
-function constructGiftOptionsStr (gifts) {
+"use strict"
+
+const constructGiftOptionsStr = (gifts) => {
   return gifts.reduce((str, gift) => {
-    return `${str}${gift.gift_name}, code: ${gift.look_up}${gift.gift_id}\n`
+    return `${str}>${gift.gift_name} -- CODE: ${gift.look_up}\n`
   }, "")
 }
+const constructAddressCodesOptionsStr = (addressCodes) => {
+  return addressCodes.reduce((str, code) => {
+    return `${str}${code}\n`
+  }, "Address Codes:\n")
+}
 
-exports.constructSignUpGiftOptionsMessages = gifts => {
-  const message = `Hey! You have successfully signed up for SurpriseHer monthly limited gift lists! Here is our curated gift selection only avalible to new signups:\n\n
-  ${constructGiftOptionsStr(gifts)} PRODUCT URL\n\n
-  to begin the order process, txt back the gift code and the day in month you'd like us to ship (e.g 3 ${gifts[0].look_up}${gifts[0].gift_id}, 18 ${gifts[1].look_up}${gifts[1].gift_id})`
+const constructSignUpGiftOptionsMessages = (userRegistered, gifts, addresses) => {
+  let message = ''
+  if( userRegistered ) {
+    const addressCodes = addresses.map( address => address.code_name)
+    message = `Hey! You have successfully signed up for SurpriseHer monthly limited gift lists! Here is our curated gift selection only avalible to new signups:
+    \n\n${constructGiftOptionsStr(gifts)}PRODUCT URL 
```

Comment:
@armandopmj es6 backticks respect newlines, so unless you actually want 3 newlines, this might be too many.